x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycle.java

```diff
@@ -137,6 +138,12 @@ protected Clock getClock() {
         }
         indexLifecycleInitialisationService.set(new IndexLifecycleService(settings, client, clusterService, threadPool,
                 getClock(), System::currentTimeMillis, xContentRegistry));
+        clusterService.addLifecycleListener(new LifecycleListener() {
```

Comment:
IndexLifecycle Plugin closes the scheduler when it itself is closed. Is this not done by the node 
where it calls close() on all the plugins?

Reply:
It does but I think we may want to stop before then. The node close mechanism calls `stop()` first, which stops the cluster service but the IndexLifecycle job can still run before the plugin is actually closed. That said, I can remove this part of the change since I don't think it will actually affect the issue at hand.